docs/source/reference/types.rst

```diff
@@ -329,3 +329,44 @@ Optional types
       True
       >>> f(None)
       False
+
+
+Type annotations
+-----------------
+
+.. function:: numba.extending.as_numba_type(py_type)
+
+   Create a Numba type corresponding to the given Python *type annotation*.
+   ``TypingError`` is raised if the type annotation can't be mapped to a Numba
+   type.  This function is meant to be used at statically compile time to
+   evaluate Python type annotations.  For runtime checking of Python objects
+   see ``typeof`` above.
+
+   For any numba type, ``as_numba_type(nb_type) == nb_type``.
+
+      >>> numba.extending.as_numba_type(int)
+      int64
+      >>> import typing  # the Python library, not the Numba one
+      >>> numba.extending.as_numba_type(typing.List[float])
+      ListType[float64]
+      >>> numba.extending.as_numba_type(numba.int32)
+      int32
+
+   ``as_numba_type`` is automatically updated to include any ``@jitclass``.
+
+      >>> @jitclass
+      ... class Counter:
+      ...     x: int
```

Comment:
Perhaps worth allowing (or requiring) the spelling
```suggestion
      ...     x: typing.Annotated[int, numba.int64]
```

Reply:
I didn't realize PEP 593 had been back-ported in `typing_extensions` already.  Awesome!  Added support for it.  See the updated docs and examples.